src/components/game/GameEngine.js

```diff
@@ -2,18 +2,52 @@ import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import shuffleArray from 'lodash.shuffle';
 import { getPathByName, getWordsForPath } from '../../db/db';
+import { useNavigate } from 'react-router-dom';
 import './GameEngine.css';
 
-const GameEngine = ({ pathName }) => {
-  const [words, setWords] = useState([]);
-  const [currentWord, setCurrentWord] = useState(null);
+const GameEngine = ({ pathName, levelIndex: initialLevelIndex }) => {
   const [currentPhase, setCurrentPhase] = useState(1);
   const [playerInput, setPlayerInput] = useState('');
   const [shuffledWord, setShuffledWord] = useState('');
-  const [wordIndex, setWordIndex] = useState(0);
   const [gameOver, setGameOver] = useState(false);
+  const [levels, setLevels] = useState([]);
+  const [currentLevel, setCurrentLevel] = useState([]);
+  const [currentWord, setCurrentWord] = useState(null);
+  const [wordIndex, setWordIndex] = useState(0);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState(null);
+  const [levelIndex, setLevelIndex] = useState(initialLevelIndex - 1);
+
+  const navigate = useNavigate();
+
+  const wordsIntoLevels = (words) => {
```

Comment:
En nyt ihan täysin ole vakuuttunut tämän logiikasta. Miten sen pitäisi toimia? Ei ole myöskään kovinkaan selkolukuista, erityisesti useamman for-loopin myötä. Jos on tuttua niin suosi funktionaalista ohjelmointia: https://www.toptal.com/javascript/functional-programming-javascript

Jos on vieras käsite, niin ei kannata tässä vaiheessa kuitenkaan käyttää sen opetteluun aikaa.

Reply:
Keskustellaan tästä tänään lisää tiimitapaamisessa. Funktionaalinen ohjelmointi ei ole ennestään tuttua, mutta tasoihin jaon logiikkaa voisi varmasti kyllä yksinkertaistaa.